src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_CodeLens.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CodeLens;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal partial class CodeAnalysisService
+    {
+        public async Task<ReferenceCount> GetReferenceCountAsync(DocumentId documentId, TextSpan textSpan, int maxResultCount, byte[] solutionChecksum)
+        {
+            using (Internal.Log.Logger.LogBlock(FunctionId.CodeAnalysisService_GetReferenceCountAsync, documentId.ProjectId.DebugName, CancellationToken))
+            {
```

Comment:
you probably want to have try/catch like the one in CalculateDiagnosticsAsync.

I need to clean up the need to have that try/catch, but for now, you need them.

catch for operationcanceledException will be gone once we move to new StreamJsonRpc which support cancellation token natively.

catch for IOException need to move into AssetSource. but until that clean up, you need try/catch.